samples/face_reid/recognition.py

```diff
@@ -0,0 +1,77 @@
+"""Recognition module."""
+import os
+import hnswlib
+
+from savant.gstreamer import Gst
+from savant.deepstream.meta.frame import NvDsFrameMeta
+from savant.deepstream.pyfunc import NvDsPyFuncPlugin
+from savant.parameter_storage import param_storage
+from savant.meta.constants import UNTRACKED_OBJECT_ID
+from samples.face_reid.utils import unpack_person_id_img_n
+
+MODEL_NAME = param_storage()['reid_model_name']
+
+
+class Recognition(NvDsPyFuncPlugin):
+    """Uses HNSW index to find nearest neighbor for each face feature vector."""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.last_match_for_track = {}
+        self.index_file_path = os.path.join(self.index_dir, 'index.bin')
+
+    def on_start(self) -> bool:
+        """Do on plugin start."""
+        self.index = hnswlib.Index(space=self.index_space, dim=self.index_dim)
+        self.index.load_index(
+            self.index_file_path, max_elements=self.index_max_elements
+        )
+        return True
+
+    def process_frame(self, buffer: Gst.Buffer, frame_meta: NvDsFrameMeta):
+        """Process frame metadata.
+
+        :param buffer: Gstreamer buffer with this frame's data.
+        :param frame_meta: This frame's metadata.
+        """
+
+        # gather all face feature vectors from this frame
+        features = []
+        for obj_meta in frame_meta.objects:
+            if obj_meta.label == 'face':
+                features.append(obj_meta.get_attr_meta(MODEL_NAME, 'feature').value)
+
+        if len(features) < 1:
```

Comment:
I think `if not features` is more pythonic)

Reply:
Fixed